lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -54,27 +42,42 @@ def create_database_with_no_switching(*args)
     end
 
     def disconnect_with_host_pooling!
-      @_cached_current_database = nil
-      @_cached_connection_object_id = nil
+      self._cached_connection_object_id = nil
       disconnect_without_host_pooling!
     end
 
+    def _host_pool_current_database
+      _ahrp_storage[:_host_pool_current_database]
```

Comment:
```suggestion
      _arhp_storage[:_host_pool_current_database]
```

I also make this mistake constantly when working in this repo but `ahrp` should be renamed `arhp`